### PR TITLE
Add gset support

### PR DESCRIPTION
**Pull request by russelldb**
This PR is not ready for merging, but I'm opening it for review. The rebase was difficult because of the refactoring out of the test code so I chose to merge the non-conflicting code and re-implement the remaining change.

I'd appreciate your eyes briefly @binarytemple-bet365

### CODE REVIEW

**Comment by binarytemple-bet365** [comment]
@russelldb -  Branch comparison:

[rdb/bet365/gsets...bet365:feature/bet365/gsets](https://github.com/basho/riak-erlang-client/compare/rdb/bet365/gsets...bet365:feature/bet365/gsets)

[/feature/bet365/gsets...basho:rdb/bet365/gsets](https://github.com/bet365/riak-erlang-client/compare/feature/bet365/gsets...basho:rdb/bet365/gsets)

Checked it out, seems the same to me.

**Comment by binarytemple-bet365** [comment]
@russelldb Well the tests  moved to their own file -  riakc_pb_socket_test.erl - and the test '     {"add item to gset, twice"' is in there as expected.

File: test/riakc_pb_socket_tests.erl [hunk 1]
```diff
@@ -1378,6 +1378,23 @@ integration_tests() ->
                      Rsp ->
                          ?debugFmt("hlls bucket is not present, skipping (~p)", [Rsp])
                  end
+             end)},
+     {"add item to gset, twice",
+      ?_test(begin
+                 riakc_test_utils:reset_riak(),
+                 {ok, Pid} = riakc_test_utils:start_link(test_ip(), test_port()),
```
Comment (binarytemple-bet365) [inline test/riakc_pb_socket_tests.erl]: @russelldb  This has been changed to no args `riakc_test_utils:start_link()`

**Comment by russelldb** [comment]
Thanks @binarytemple-bet365 

I just made some small changes. Tests pass for me now, so I'm plus one on this

**Comment by russelldb** [comment]
+1

**Comment by russelldb** [comment]
of course needs this PB branch https://github.com/basho/riak_pb/pull/209

**Comment by russelldb** [comment]
Closing. Missed the 2.2 boat, going in 2.3, see https://github.com/basho/riak-erlang-client/pull/327

**Pull request closed without merging**